### DISCUSSION TITLE
Cannot build fastRTPS for ubuntu 22.04

### DISCUSSION THREAD

**Discussion by kyuhyong1** [eProsima/Fast-DDS#4242] [Q&A]
Tried to follow below instruction on ubuntu 22.04 with ros2-humble installed system.
https://docs.px4.io/v1.13/en/dev_setup/fast-dds-installation.html

During make I am keep getting below error.
```
Fast-DDS/src/cpp/security/authentication/PKIDH.cpp:861:37: error: invalid conversion from ‘const ec_key_st*’ to ‘EC_KEY*’ {aka ‘ec_key_st*’} [-fpermissive]
  861 |                 EVP_PKEY_get0_EC_KEY(dhkey);
      |                 ~~~~~~~~~~~~~~~~~~~~^~~~~~~
      |                                     |
      |                                     const ec_key_st*
```
Why this error happens? Any idea?

**Comment by JesusPoderoso** [comment]
Hi @kyuhyong1, thanks for using Fast DDS.
In ROS 2 Humble workspace, please make sure you are using a Fast DDS v2.6.x version and Open SSL v1.1.1 version.
This error may be caused by Open SSL v3.
